### PR TITLE
Admin: Adds a legal document button + fixes mock data

### PR DESCRIPTION
**Pull request by sarthaktexas**
## Summary of the problem
Makes a more efficient workflow for admins.

## Describe your changes
This pr:
1. Releases a new feature flag `:admin_legal_documents_2024_02_05` for a feature (that allows them to go directly to legal documents) so that admins that don't need this feature don't get it.
image

2. Refactors event feature flags partial
3. Fixes mock data so there are only 2 mock cards with two different states. Screenshot 2024-02-05 at 12 26 47 PM
4. Admin tools **do not** show up when mock data is visible. This is so during onboarding calls they do not see the tools we use.

|       | Ledger                                                                                                                                                  | Users page                                                                                                                                                  |
|-------|---------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|
| In playground mode | Screenshot 2024-02-05 at 12 24 50 PM | Screenshot 2024-02-05 at 12 24 40 PM |
| Out of playground mode | image                                | Screenshot 2024-02-05 at 12 25 02 PM |

### CODE REVIEW

File: app/views/events/_nav.html.erb [hunk 1]
```diff
@@ -104,6 +104,13 @@
         tooltip: "Commonly asked questions and how-tos",
         icon: "info",
         selected: local_assigns[:selected] == :documentation %>
+      <% admin_tools("m0 p0 overflow-visible w-fit") do %>
+        <%= dock_item "Legal Documents",
+        "/#{@event.slug}/documents",
```
Comment (sampoder) [inline app/views/events/_nav.html.erb]:
```suggestion
        event_documents_path(@event),
```

Comment (sarthaktexas) [reply app/views/events/_nav.html.erb]: thanks, i was wondering how to find these but eventually gave up

File: app/assets/stylesheets/components/_features.scss [hunk 2]
```diff
@@ -43,6 +43,11 @@ html[data-dark='true'] {
   background-size: 8em;
 }
 
+#legal-documents-tab {
+  margin: 0;
+  background-color: rgba(map-get($palette, warning), 0.0625);
+}
+
```
Comment (garyhtou) [inline app/assets/stylesheets/components/_features.scss]: There may be a better place for this CSS. this file is for styling feature previews

File: app/views/events/_nav.html.erb [hunk 3]
```diff
@@ -104,6 +106,15 @@
         tooltip: "Commonly asked questions and how-tos",
         icon: "info",
         selected: local_assigns[:selected] == :documentation %>
+      <% if Flipper.enabled?(:admin_legal_documents_2024_02_05, current_user) && !show_mock_data? %>
```
Comment (garyhtou) [inline app/views/events/_nav.html.erb]: We should just enable this for all admins. Flipper flags should be short-lived. We want to eventually get rid of them

**Review by garyhtou — COMMENTED** [review]
notes from meeting:

- events partial is still needed. But, we can refactor them in a partial
- Split these separate PRs
- Mock data check should maybe be included inside `admin_tools`

**Comment by sampoder** [comment]
I think I already have a PR in for feature flags, not sure if there's crossover here: https://github.com/hackclub/hcb/pull/4883

File: app/assets/stylesheets/components/_features.scss (hunk 2 above)
Comment (sarthaktexas) [reply app/assets/stylesheets/components/_features.scss]: that's what this is

**Pull request closed without merging**